### PR TITLE
Code Review Preparation, Tic-Tac-Toe Milestone 2

### PR DESCRIPTION
**Pull request by jamezjaz**
On this Milestone, we defined the following methods:
- display_board
- players_turn
- flip_player
- play_game.
- We also displayed players' input on the board

### CODE REVIEW

File: bin/main [hunk 1]
```diff
@@ -1,3 +1,56 @@
 #!/usr/bin/env ruby
+# rubocop: disable Style/GlobalVars:
+# rubocop: disable Style/WordArray:
```
Comment (maelfosso) [inline bin/main]: No, please remove it.

File: bin/main [hunk 2]
```diff
@@ -1,3 +1,56 @@
...
+# check tie
+# flip player
+
+$board = [
+  '_', '_', '_',
+  '_', '_', '_',
+  '_', '_', '_'
+]
+$game_on_going = true
+$current_player = 'X'
```
Comment (maelfosso) [inline bin/main]: The global variables are not allowed in this project.
Be focused on OOP

File: bin/main [hunk 3]
```diff
@@ -1,3 +1,56 @@
...
+                    end
+end
+
+def play_game
+  while $game_on_going
+    players_turn
+    flip_player
+    display_board
+  end
+end
```
Comment (maelfosso) [inline bin/main]: It's running indefinitely.

Comment (maelfosso) [reply bin/main]: I suggest you change it like this
```rb
game_on = true

# initial instructions for player

while game_on
  # loop for each move
  
  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello @Emmanuelaaron @jamezjaz 

Good introduction to this milestone. However, allow me to precise the goal of this milestone

* [ ] In this step, not everything must work, the point is to make sure that you thought about everything listed below. They might do it only with inline comments. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/logic.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [ ] Basic game flow is prepared: ([screenshot hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/logic.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
  * [ ] `[OPTIONAL]` game ask players for their names
  * [ ] `[REQUIRED]` game tells which player turn it is
  * [ ] `[REQUIRED]` game asks player to select from available moves
  * [ ] `[REQUIRED]` game informs player if selected move is invalid
  * [ ] `[REQUIRED]` game displays board after player move
  * [ ] `[REQUIRED]` game informs player if selected move is a winning move
  * [ ] `[REQUIRED]` game informs player if selected move is a draw move
  * [ ] `[REQUIRED]` game repeats all actions for next player's move

So, please, be sure to complete the checklist and then submit a new code review request.
Thanks

Mael FOSSO

**Comment by Dandush03** [comment]
# Status: :green_circle: Project Approved :green_circle:

## Hey Guys :wave:

### You did a great job implementing all previously requirement. please read carefully the next observation marked as OPTIONAL

[__OPTIONAL__]

As I said before, you guys meet all previews requirement but what I'm going to describe ahead is a new configuration of Stickler that will be implementing ASAP in our pathrwright.  and I want to make it clear, this is a new upgrade feature of stickler witch will have to implement ASAP. 

Saying that. Rubocop gem in Stickler is not running through files that don't have the Ruby extension (*.rb) therefore I will ask you guys to update your code with the new implementation, renaming the main file to main.rb despite that the pathwright is asking you to use the bin file.

I will make sure that any future review will check for this issue so it will not cause any future inconvenience. I must inform you guys that as far as now your file is not passing most of Rubocop flags, so I hope you take a step forward making these changes.

# Happy Coding & GL on your next Project :wink:

## Daniel Laloush :nerd_face:

**Pull request merged**